cmaa.sevsnp.attest.sample/Program.cs

```diff
@@ -0,0 +1,422 @@
+            {
+                foreach (var key in jsonResponse.keys)
+                {
+                    var certBase64 = key?.x5c[0]?.ToString();
+                    if (!string.IsNullOrEmpty(certBase64))
+                    {
+                        var certBytes = Convert.FromBase64String(certBase64);
+                        certificates.Add(new X509Certificate2(certBytes));
+                    }
+                }
+            }
+            return certificates.Count > 0 ? certificates : null;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(">>> Certificate Retrieval Error: " + ex.Message);
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Verifies the digital signature of a JWT token using the provided X.509 certificate.
+    /// This ensures that the token has been signed by a trusted entity and has not been tampered with.
+    /// </summary>
+    /// <param name="jwtToken">The JWT token whose signature needs to be verified.</param>
+    /// <param name="certificate">The X.509 certificate containing the public key used for signature verification.</param>
+    /// <returns>
+    /// Returns <c>true</c> if the token's signature is valid and was signed by the expected issuer. 
+    /// Returns <c>false</c> if the signature verification fails, the certificate's public key cannot be extracted, 
+    /// or an exception occurs during validation.
+    /// </returns>
+    private static bool VerifyTokenSignature(JwtSecurityToken jwtToken, X509Certificate2 certificate)
+    {
+        try
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var rsa = certificate.GetRSAPublicKey();
+            if (rsa == null)
+            {
+                Console.WriteLine(">>> RSA public key extraction failed from certificate: " + certificate.Subject);
+                return false;
+            }
+
+            // Normalize certificate issuer to match JWT Issuer
+            string normalizedCertIssuer = certificate.Issuer.Replace("CN=", "").Trim();
+
+            var validationParameters = new TokenValidationParameters
+            {
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = new RsaSecurityKey(rsa),
+                ValidateIssuer = true,
+                ValidIssuer = normalizedCertIssuer,
+                ValidateAudience = false,
+                ValidateLifetime = false
+            };
+
+            tokenHandler.ValidateToken(jwtToken.RawData, validationParameters, out _);
+            Console.WriteLine("VerifyTokenSignature - Token signature is valid.");
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(">>> Token signature validation failed: " + ex.Message);
+            return false;
+        }
+    }
+
+    private static bool VerifyTokenSignatureManually(JwtSecurityToken jwtToken, X509Certificate2 certificate)
```

Comment:
Is this still needed?

Reply:
not really, I left it for the review to decide which method to go with